src/devices/machine/ldp1450hle.cpp

```diff
@@ -91,6 +93,244 @@ sony_ldp1450hle_device::sony_ldp1450hle_device(const machine_config &mconfig, co
+//  overlay_draw_group - draw a single group of
+//  characters
+//-------------------------------------------------
+
+void sony_ldp1450hle_device::overlay_fill(bitmap_yuy16 &bitmap, uint8_t yval, uint8_t cr, uint8_t cb)
+{
+	uint16_t color0 = (yval << 8) | cb;
+	uint16_t color1 = (yval << 8) | cr;
+
+	// write 32 bits of color (2 pixels at a time)
+	for (int y = 0; y < bitmap.height(); y++)
+	{
+		uint16_t *dest = &bitmap.pix(y);
+		for (int x = 0; x < bitmap.width() / 2; x++)
+		{
+			*dest++ = color0;
+			*dest++ = color1;
+		}
+	}
+}
+
+void sony_ldp1450hle_device::overlay_draw_group(bitmap_yuy16 &bitmap, const uint8_t *text, int start, int xstart, int ystart, int mode)
+{
+	u8 char_width = text_size[m_user_index_mode & 0x03];
+
+	u8 char_height = text_size[(m_user_index_mode >> 2) & 0x03];
+
+	float xstart_normalised = (bitmap.width()/2/ 64) * xstart;
+	float ystart_normalised = (bitmap.height()/ 64) * ystart;
+
+	if (m_user_index_mode & 0x80) //blue screen
+	{
+		overlay_fill(bitmap, 0x28, 0x6d, 0xf0);
+	}
+
+	u8 count;
+	if (m_user_index_mode & 0x10) // 3 line mode
+	{
+		LOGMASKED(LOG_COMMANDS, "3 Line\n");
+		int x = 0;
+		for (int y = 0; y < 3; y++)
+		{
+			bool eos = false;
+			for (int char_idx = x; char_idx < (x+10); char_idx++)
+			{
+				if (text[char_idx] == 0x1a || eos)
+				{
+					eos = true;
+					x -= 9;// adjust pointer to account for the + 10 we do automatically
+					break;
+				}
+				else
+				{
+					overlay_draw_char(bitmap, text[char_idx], xstart_normalised + (char_width  * (char_idx %10)), ystart_normalised + (y*char_height), char_width, char_height);
+				}
+			}
+			x += 10;
+		}
+	}
+	else
+	{
+		count = 0x20 - start;
+		LOGMASKED(LOG_COMMANDS, "1 Line\n");
+
+		for (int x = start; x < count; x++)
+		{
+			if (text[x] == 0x1a)
+			{
+				break;
+			}
+			else
+			{
+				overlay_draw_char(bitmap, text[x], xstart_normalised + ((char_width) * x), ystart_normalised, char_width, char_height);
+			}
+		}
+	}
+}	
+
+
+//-------------------------------------------------
+//  overlay_draw_char - draw a single character
+//  of the text overlay
+//-------------------------------------------------
+
+void sony_ldp1450hle_device::overlay_draw_char(bitmap_yuy16 &bitmap, uint8_t ch, float xstart, int ystart, int char_width, int char_height)
+{
+
+	// m_user_index_mode >> 5 & 0x04: 0,2 = normal, 1 = 1px shadow, 3 = grey box 
+	uint16_t black = 0x0080;
+
+	u8 modeval= (m_user_index_mode >> 5) & 0x04;
+	
+	for (u32 y = 0; y < char_height; y++)
+	{
+		for (u8 x = 0; x < char_width; x++)
+		{
+			u32 xmin = xstart + x;
+			for (u32 yy = 0; yy < OVERLAY_PIXEL_HEIGHT; yy++)
+			{
+
+				for (u32 xx = 0; xx < OVERLAY_PIXEL_WIDTH; xx++)
+				{
+					if (modeval==0x03)
+					{
+						//fill with grey	
+					}
+
+					if (m_osd_font[ch].pix(y,x) != black)
+					{	
+						bitmap.pix(ystart + (y + 1) * OVERLAY_PIXEL_HEIGHT + yy, xmin+xx) = m_osd_font[ch].pix(y,x);
+					}
+				}
+			}
+		}
+	}
```

Comment:
It’s impossible for the `modeval == 0x03` test to pass, because you assigned it something `& 0x04` above, and `0x04 & 0x03 == 0`.